Sources/MapboxMaps/Viewport/States/FollowPuck/FollowPuckViewportState.swift

```diff
@@ -44,10 +44,27 @@ extension FollowPuckViewportState: ViewportState {
         guard updatingCameraCancelable == nil else {
             return
         }
-        updatingCameraCancelable = dataSource.observe { [mapboxMap] cameraOptions in
-            mapboxMap.setCamera(to: cameraOptions)
+        var animationStarted = false
+        var animationComplete = false
+
+        let compositeCancelable = CompositeCancelable()
+
+        updatingCameraCancelable = compositeCancelable
+
+        compositeCancelable.add(dataSource.observe { [mapboxMap, cameraAnimationsManager, options] cameraOptions in
+            if animationComplete {
+                mapboxMap.setCamera(to: cameraOptions)
+            } else if !animationStarted {
+                animationStarted = true
+                compositeCancelable.add(cameraAnimationsManager.ease(
+                    to: cameraOptions,
+                    duration: options.animationDuration,
+                    curve: .linear) { _ in
+                        animationComplete = true
+                    }!)
```

Comment:
This force-unwrap is "safe" since `ease(to:…` never returns `nil`. We have a ticket to make the return type non-optional in v11.

Reply:
Good that you highlighted this! The exclamation mark is quite hard to notice there :) It will take some time to get v11 out, maybe let's stick to the safer side here meanwhile and unwrap this. There will be a compiler error anyway when the return type is non-optional so it won't be forgotten.

Reply:
How do you propose handling the case where it returns nil?

Reply:
I have an idea for how to clean this up. Will push shortly…